File: expense-tracker/app/src/main/java/com/farhan/paisatrack/ui/screens/DashboardScreen.kt
package com.farhan.paisatrack.ui.screens

import androidx.compose.animation.AnimatedVisibility
import androidx.compose.animation.core.animateFloatAsState
import androidx.compose.animation.core.tween
import androidx.compose.animation.fadeIn
import androidx.compose.animation.fadeOut
import androidx.compose.foundation.ExperimentalFoundationApi
import androidx.compose.foundation.background
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.width
import androidx.compose.foundation.lazy.LazyColumn
import androidx.compose.foundation.lazy.items
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.automirrored.filled.TrendingDown
import androidx.compose.material.icons.automirrored.filled.TrendingUp
import androidx.compose.material.icons.filled.CheckCircle
import androidx.compose.material.icons.filled.Close
import androidx.compose.material.icons.filled.Settings
import androidx.compose.material3.Card
import androidx.compose.material3.CardDefaults
import androidx.compose.material3.Icon
import androidx.compose.material3.IconButton
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.material3.TextButton
import androidx.compose.runtime.Composable
import androidx.compose.runtime.getValue
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Brush
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import androidx.lifecycle.compose.collectAsStateWithLifecycle
import com.farhan.paisatrack.data.AccountType
import com.farhan.paisatrack.data.TxnType
import com.farhan.paisatrack.ui.MainViewModel
import com.farhan.paisatrack.ui.components.IconBadge
import com.farhan.paisatrack.ui.components.SectionCard
import com.farhan.paisatrack.ui.theme.ExpenseRed
import com.farhan.paisatrack.ui.theme.IncomeGreen
import com.farhan.paisatrack.ui.theme.Violet
import com.farhan.paisatrack.ui.theme.VioletDark
import com.farhan.paisatrack.util.Format
import com.farhan.paisatrack.util.IconMap

@OptIn(ExperimentalFoundationApi::class)
@Composable
fun DashboardScreen(
    vm: MainViewModel,
    onAddTxn: () -> Unit,
    onOpenTxn: (Long) -> Unit,
    onSeeAllTxns: () -> Unit,
    onSeePeople: () -> Unit,
    onOpenSettings: () -> Unit,
    onPayCard: (Long) -> Unit,
    onOpenAccount: (Long) -> Unit
) {
    val dash by vm.dashboard.collectAsStateWithLifecycle()
    val txns by vm.confirmedTxns.collectAsStateWithLifecycle()
    val pending by vm.pendingTxns.collectAsStateWithLifecycle()
    val accounts by vm.accounts.collectAsStateWithLifecycle()
    val categories by vm.allCategories.collectAsStateWithLifecycle()

    val catById = categories.associateBy { it.id }
    val accById = accounts.associateBy { it.id }

    LazyColumn(
        modifier = Modifier.fillMaxWidth(),
        contentPadding = androidx.compose.foundation.layout.PaddingValues(16.dp, 8.dp, 16.dp, 96.dp),
        verticalArrangement = Arrangement.spacedBy(14.dp)
    ) {
        item {
            Row(verticalAlignment = Alignment.CenterVertically) {
                Column(Modifier.weight(1f)) {
                    Text(
                        "Hello, Farhan 👋",
                        style = MaterialTheme.typography.titleMedium,
                        color = MaterialTheme.colorScheme.onSurfaceVariant
                    )
                    Text("Your money at a glance", style = MaterialTheme.typography.headlineSmall, fontWeight = FontWeight.Bold)
                }
                IconButton(onClick = onOpenSettings) {
                    Icon(Icons.Filled.Settings, contentDescription = "Settings")
                }
            }
        }

        item { NetWorthCard(dash) }

        item {
            Row(horizontalArrangement = Arrangement.spacedBy(12.dp)) {
                StatTile(
                    modifier = Modifier.weight(1f),
                    label = "To receive",
                    value = Format.money(dash.toReceive),
                    color = IncomeGreen,
                    icon = Icons.AutoMirrored.Filled.TrendingUp,
                    onClick = onSeePeople
                )
                StatTile(
                    modifier = Modifier.weight(1f),
                    label = "To pay",
                    value = Format.money(dash.toPay),
                    color = ExpenseRed,
                    icon = Icons.AutoMirrored.Filled.TrendingDown,
                    onClick = onSeePeople
                )
            }
        }

        item {
            Row(horizontalArrangement = Arrangement.spacedBy(12.dp)) {
                StatTile(
                    modifier = Modifier.weight(1f),
                    label = "This month spent",
                    value = Format.money(dash.monthExpense),
                    color = ExpenseRed,
                    icon = Icons.AutoMirrored.Filled.TrendingDown
                )
                StatTile(
                    modifier = Modifier.weight(1f),
                    label = "This month in",
                    value = Format.money(dash.monthIncome),
                    color = IncomeGreen,
                    icon = Icons.AutoMirrored.Filled.TrendingUp
                )
            }
        }

        if (pending.isNotEmpty()) {
            item {
                Text("Detected from SMS — confirm", style = MaterialTheme.typography.titleMedium, fontWeight = FontWeight.SemiBold)
            }
            items(pending, key = { "p${it.id}" }) { t ->
                PendingTxnCard(
                    modifier = Modifier.animateItemPlacement(),
                    title = if (t.merchant.isNotBlank()) t.merchant else "Unknown",
                    subtitle = "${Format.dateTime(t.dateTime)}  ·  tap to categorize",
                    amount = t.amount,
                    isIncome = t.type == TxnType.INCOME,
                    onConfirm = { onOpenTxn(t.id) },
                    onDismiss = { vm.dismissPending(t) }
                )
            }
        }

        item {
            SectionHeader("Accounts", actionLabel = null, onAction = null)
        }
        items(dash.balances, key = { "acc${it.account.id}" }) { ab ->
            AccountRow(
                modifier = Modifier.animateItemPlacement(),
                accountId = ab.account.id,
                name = ab.account.name,
                icon = IconMap.icon(ab.account.icon),
                color = IconMap.parseColor(ab.account.colorHex),
                balance = ab.balance,
                isCredit = ab.account.type == AccountType.CREDIT_CARD,
                onPayCard = onPayCard,
                onClick = onOpenAccount
            )
        }

        item {
            SectionHeader("Recent activity", actionLabel = "See all", onAction = onSeeAllTxns)
        }
        if (txns.isEmpty()) {
            item {
                SectionCard {
                    Text(
                        "No transactions yet. Tap + to add your first one, or scan your SMS inbox in Settings.",
                        style = MaterialTheme.typography.bodyMedium,
                        color = MaterialTheme.colorScheme.onSurfaceVariant
                    )
                }
            }
        } else {
            items(txns.take(6), key = { "t${it.id}" }) { t ->
                val cat = t.categoryId?.let { catById[it] }
                TransactionRow(
                    modifier = Modifier.animateItemPlacement(),
                    title = when {
                        t.contactName != null -> t.contactName!!
                        t.merchant.isNotBlank() -> t.merchant
                        cat != null -> cat.name
                        else -> t.type.name.lowercase().replaceFirstChar { it.uppercase() }
                    },
                    subtitle = buildString {
                        append(cat?.name ?: t.type.name.lowercase().replaceFirstChar { it.uppercase() })
                        accById[t.accountId]?.let { append(" · ${it.name}") }
                    },
                    note = t.note,
                    amount = t.amount,
                    type = t.type,
                    icon = IconMap.icon(cat?.icon ?: "category"),
                    color = IconMap.parseColor(cat?.colorHex ?: "#636E72"),
                    dateText = Format.day(t.dateTime),
                    onClick = { onOpenTxn(t.id) }
                )
            }
        }
    }
}

@Composable
private fun NetWorthCard(dash: com.farhan.paisatrack.ui.DashboardState) {
    val animatedLiquid by animateFloatAsState(
        targetValue = dash.liquidTotal.toFloat(),
        animationSpec = tween(700),
        label = "liquid"
    )
    Card(
        modifier = Modifier.fillMaxWidth(),
        shape = RoundedCornerShape(24.dp),
        colors = CardDefaults.cardColors(containerColor = Color.Transparent)
    ) {
        Box(
            Modifier
                .background(Brush.linearGradient(listOf(Violet, VioletDark)))
                .padding(20.dp)
        ) {
            Column {
                Text("Available balance (cash + banks)", color = Color.White.copy(alpha = 0.85f), fontSize = 13.sp)
                Spacer(Modifier.height(6.dp))
                Text(
                    Format.money(animatedLiquid.toDouble()),
                    color = Color.White,
                    fontSize = 34.sp,
                    fontWeight = FontWeight.Bold
                )
                Spacer(Modifier.height(16.dp))
                Row(Modifier.fillMaxWidth(), horizontalArrangement = Arrangement.SpaceBetween) {
                    MiniStat("Net worth", Format.moneyShort(dash.netWorth))
                    MiniStat("Invested", Format.moneyShort(dash.investedValue))
                    MiniStat("Card due", Format.moneyShort(dash.creditOutstanding))
                }
            }
        }
    }
}

@Composable
private fun MiniStat(label: String, value: String) {
    Column {
        Text(label, color = Color.White.copy(alpha = 0.8f), fontSize = 12.sp)
        Text(value, color = Color.White, fontWeight = FontWeight.SemiBold, fontSize = 16.sp)
    }
}

@Composable
private fun StatTile(
    modifier: Modifier = Modifier,
    label: String,
    value: String,
    color: Color,
    icon: androidx.compose.ui.graphics.vector.ImageVector,
    onClick: (() -> Unit)? = null
) {
    Card(
        modifier = modifier.then(if (onClick != null) Modifier.clickable { onClick() } else Modifier),
        shape = RoundedCornerShape(18.dp),
        colors = CardDefaults.cardColors(containerColor = MaterialTheme.colorScheme.surface),
        elevation = CardDefaults.cardElevation(defaultElevation = 1.dp)
    ) {
        Column(Modifier.padding(14.dp)) {
            IconBadge(icon = icon, tint = color, size = 36)
            Spacer(Modifier.height(10.dp))
            Text(label, fontSize = 12.sp, color = MaterialTheme.colorScheme.onSurfaceVariant)
            Text(value, fontWeight = FontWeight.Bold, fontSize = 18.sp)
        }
    }
}

@Composable
fun SectionHeader(title: String, actionLabel: String?, onAction: (() -> Unit)?) {
    Row(
        Modifier.fillMaxWidth().padding(top = 4.dp),
        horizontalArrangement = Arrangement.SpaceBetween,
        verticalAlignment = Alignment.CenterVertically
    ) {
        Text(title, style = MaterialTheme.typography.titleMedium, fontWeight = FontWeight.SemiBold)
        if (actionLabel != null && onAction != null) {
            Text(
                actionLabel,
                color = MaterialTheme.colorScheme.primary,
                fontWeight = FontWeight.SemiBold,
                modifier = Modifier.clickable { onAction() }
            )
        }
    }
}

@Composable
private fun AccountRow(
    modifier: Modifier = Modifier,
    accountId: Long,
    name: String,
    icon: androidx.compose.ui.graphics.vector.ImageVector,
    color: Color,
    balance: Double,
    isCredit: Boolean,
    onPayCard: (Long) -> Unit,
    onClick: (Long) -> Unit
) {
    SectionCard(modifier = modifier.clickable { onClick(accountId) }) {
        Row(verticalAlignment = Alignment.CenterVertically) {
            IconBadge(icon = icon, tint = color)
            Spacer(Modifier.width(12.dp))
            Column(Modifier.weight(1f)) {
                Text(name, fontWeight = FontWeight.SemiBold)
                Text(
                    if (isCredit) "Outstanding" else "Balance",
                    fontSize = 12.sp,
                    color = MaterialTheme.colorScheme.onSurfaceVariant
                )
            }
            Column(horizontalAlignment = Alignment.End) {
                Text(
                    Format.money(if (isCredit) -balance else balance),
                    fontWeight = FontWeight.Bold,
                    color = if (isCredit && balance > 0) ExpenseRed else MaterialTheme.colorScheme.onSurface
                )
                if (isCredit) {
                    TextButton(
                        onClick = { onPayCard(accountId) },
                        contentPadding = androidx.compose.foundation.layout.PaddingValues(horizontal = 8.dp, vertical = 0.dp)
                    ) {
                        Text("Pay bill", fontSize = 13.sp)
                    }
                }
            }
        }
    }
}

@Composable
private fun PendingTxnCard(
    modifier: Modifier = Modifier,
    title: String,
    subtitle: String,
    amount: Double,
    isIncome: Boolean,
    onConfirm: () -> Unit,
    onDismiss: () -> Unit
) {
    Card(
        modifier = modifier.fillMaxWidth().clickable { onConfirm() },
        shape = RoundedCornerShape(18.dp),
        colors = CardDefaults.cardColors(containerColor = MaterialTheme.colorScheme.primaryContainer)
    ) {
        Row(Modifier.padding(14.dp), verticalAlignment = Alignment.CenterVertically) {
            Column(Modifier.weight(1f)) {
                Text(title, fontWeight = FontWeight.SemiBold, color = MaterialTheme.colorScheme.onPrimaryContainer)
                Text(subtitle, fontSize = 12.sp, color = MaterialTheme.colorScheme.onPrimaryContainer.copy(alpha = 0.8f))
            }
            Text(
                (if (isIncome) "+" else "-") + Format.money(amount),
                fontWeight = FontWeight.Bold,
                color = if (isIncome) IncomeGreen else ExpenseRed
            )
            IconButton(onClick = onConfirm) {
                Icon(Icons.Filled.CheckCircle, contentDescription = "Confirm", tint = IncomeGreen)
            }
            IconButton(onClick = onDismiss) {
                Icon(Icons.Filled.Close, contentDescription = "Dismiss", tint = MaterialTheme.colorScheme.onPrimaryContainer)
            }
        }
    }
}

File: farhans-wallet-flutter/lib/features/dashboard/dashboard_screen.dart
import 'package:flutter/material.dart';
import 'package:flutter_riverpod/flutter_riverpod.dart';

import '../../core/theme/tokens.dart';
import '../../core/utils/format.dart';
import '../../core/utils/icon_map.dart';
import '../../core/widgets/bits.dart';
import '../../core/widgets/buttons.dart';
import '../../core/widgets/glass.dart';
import '../../data/models.dart';
import '../../state/wallet_state.dart';
import '../credit/pay_credit_card_screen.dart';
import '../settings/account_detail_screen.dart';
import '../settings/settings_screen.dart';
import '../transactions/add_edit_transaction_screen.dart';
import '../transactions/transaction_tile.dart';

class DashboardScreen extends ConsumerWidget {
  const DashboardScreen({super.key});

  @override
  Widget build(BuildContext context, WidgetRef ref) {
    final t = context.tokens;
    final async = ref.watch(walletProvider);

    return async.when(
      loading: () => const Center(child: CircularProgressIndicator()),
      error: (e, _) => Center(child: Text('Error: $e', style: TextStyle(color: t.danger))),
      data: (data) => _Content(data: data),
    );
  }
}

class _Content extends ConsumerWidget {
  final WalletData data;
  const _Content({required this.data});

  @override
  Widget build(BuildContext context, WidgetRef ref) {
    final t = context.tokens;
    final recent = data.confirmed.take(6).toList();

    return ListView(
      padding: const EdgeInsets.fromLTRB(Insets.lg, Insets.md, Insets.lg, 120),
      children: [
        Row(
          children: [
            Expanded(
              child: Column(
                crossAxisAlignment: CrossAxisAlignment.start,
                children: [
                  Text('Hello, Farhan 👋', style: context.text.bodyMedium?.copyWith(color: t.textMid)),
                  const SizedBox(height: 2),
                  Text('Your money', style: context.text.headlineMedium),
                ],
              ),
            ),
            GlassIconButton(
              icon: Icons.settings_rounded,
              onPressed: () => Navigator.push(
                  context, MaterialPageRoute(builder: (_) => const SettingsScreen())),
            ),
          ],
        ),
        const SizedBox(height: Insets.lg),
        _BalanceCard(data: data),
        const SizedBox(height: Insets.md),
        Row(
          children: [
            Expanded(
              child: _StatTile(
                label: 'To receive',
                value: Money.format(data.toReceive),
                icon: Icons.south_west_rounded,
                color: t.success,
              ),
            ),
            const SizedBox(width: Insets.sm),
            Expanded(
              child: _StatTile(
                label: 'To pay',
                value: Money.format(data.toPay),
                icon: Icons.north_east_rounded,
                color: t.danger,
              ),
            ),
          ],
        ),
        const SizedBox(height: Insets.sm),
        Row(
          children: [
            Expanded(
              child: _StatTile(
                label: 'Spent this month',
                value: Money.format(data.monthExpense),
                icon: Icons.trending_down_rounded,
                color: t.danger,
              ),
            ),
            const SizedBox(width: Insets.sm),
            Expanded(
              child: _StatTile(
                label: 'Income this month',
                value: Money.format(data.monthIncome),
                icon: Icons.trending_up_rounded,
                color: t.success,
              ),
            ),
          ],
        ),
        if (data.pending.isNotEmpty) ...[
          const SizedBox(height: Insets.lg),
          const SectionHeader('Detected from SMS'),
          const SizedBox(height: Insets.xs),
          ...data.pending.map((p) => Padding(
                padding: const EdgeInsets.only(bottom: Insets.xs),
                child: _PendingCard(
                  txn: p,
                  onConfirm: () => Navigator.push(context,
                      MaterialPageRoute(builder: (_) => AddEditTransactionScreen(txn: p))),
                  onDismiss: () => ref.read(walletProvider.notifier).dismissPending(p),
                ),
              )),
        ],
        const SizedBox(height: Insets.lg),
        const SectionHeader('Accounts'),
        const SizedBox(height: Insets.xs),
        ...data.balances.map((b) => Padding(
              padding: const EdgeInsets.only(bottom: Insets.xs),
              child: _AccountRow(balance: b),
            )),
        const SizedBox(height: Insets.md),
        SectionHeader('Recent activity',
            actionLabel: recent.isEmpty ? null : null, onAction: null),
        const SizedBox(height: Insets.xs),
        if (recent.isEmpty)
          GlassCard(
            child: Text(
              'No transactions yet. Tap + to add one, or scan your SMS inbox in Settings.',
              style: context.text.bodyMedium?.copyWith(color: t.textMid),
            ),
          )
        else
          ...recent.map((tx) => Padding(
                padding: const EdgeInsets.only(bottom: Insets.xs),
                child: TransactionTile(
                  txn: tx,
                  data: data,
                  trailingDate: Dates.day(tx.dateTime),
                  onTap: () => Navigator.push(context,
                      MaterialPageRoute(builder: (_) => AddEditTransactionScreen(txn: tx))),
                ),
              )),
      ],
    );
  }
}

class _BalanceCard extends StatelessWidget {
  final WalletData data;
  const _BalanceCard({required this.data});

  @override
  Widget build(BuildContext context) {
    final t = context.tokens;
    return Container(
      padding: const EdgeInsets.all(Insets.lg),
      decoration: BoxDecoration(
        gradient: LinearGradient(
          colors: [t.accentA, t.accentB],
          begin: Alignment.topLeft,
          end: Alignment.bottomRight,
        ),
        borderRadius: BorderRadius.circular(Corners.lg),
        boxShadow: [
          BoxShadow(color: t.accentB.withValues(alpha: 0.4), blurRadius: 28, offset: const Offset(0, 14)),
        ],
      ),
      child: Column(
        crossAxisAlignment: CrossAxisAlignment.start,
        children: [
          Text('Available balance',
              style: TextStyle(color: Colors.white.withValues(alpha: 0.85), fontSize: 13)),
          const SizedBox(height: 6),
          AnimatedMoney(
            data.liquidTotal,
            style: const TextStyle(
                color: Colors.white, fontSize: 36, fontWeight: FontWeight.w800, letterSpacing: -1),
          ),
          const SizedBox(height: Insets.lg),
          Row(
            mainAxisAlignment: MainAxisAlignment.spaceBetween,
            children: [
              _Mini(label: 'Net worth', value: Money.short(data.netWorth)),
              _Mini(label: 'Invested', value: Money.short(data.investedValue)),
              _Mini(label: 'Card due', value: Money.short(data.creditOutstanding)),
            ],
          ),
        ],
      ),
    );
  }
}

class _Mini extends StatelessWidget {
  final String label;
  final String value;
  const _Mini({required this.label, required this.value});

  @override
  Widget build(BuildContext context) {
    return Column(
      crossAxisAlignment: CrossAxisAlignment.start,
      children: [
        Text(label, style: TextStyle(color: Colors.white.withValues(alpha: 0.8), fontSize: 12)),
        const SizedBox(height: 2),
        Text(value, style: const TextStyle(color: Colors.white, fontWeight: FontWeight.w700, fontSize: 15)),
      ],
    );
  }
}

class _StatTile extends StatelessWidget {
  final String label;
  final String value;
  final IconData icon;
  final Color color;
  const _StatTile({required this.label, required this.value, required this.icon, required this.color});

  @override
  Widget build(BuildContext context) {
    final t = context.tokens;
    return GlassCard(
      padding: const EdgeInsets.all(Insets.md),
      child: Column(
        crossAxisAlignment: CrossAxisAlignment.start,
        children: [
          IconBadge(icon: icon, color: color, size: 38),
          const SizedBox(height: Insets.sm),
          Text(label, style: context.text.bodySmall?.copyWith(color: t.textMid)),
          const SizedBox(height: 2),
          Text(value, style: context.text.titleMedium),
        ],
      ),
    );
  }
}

class _AccountRow extends StatelessWidget {
  final AccountBalance balance;
  const _AccountRow({required this.balance});

  @override
  Widget build(BuildContext context) {
    final t = context.tokens;
    final a = balance.account;
    final isCard = a.type == AccountType.creditCard;
    final shown = isCard ? -balance.balance : balance.balance;
    return GlassCard(
      onTap: () => Navigator.push(
          context, MaterialPageRoute(builder: (_) => AccountDetailScreen(accountId: a.id!))),
      padding: const EdgeInsets.all(Insets.sm),
      radius: Corners.md,
      child: Row(
        children: [
          IconBadge(icon: AppIcons.of(a.icon), color: AppIcons.parseColor(a.colorHex)),
          const SizedBox(width: Insets.sm),
          Expanded(
            child: Column(
              crossAxisAlignment: CrossAxisAlignment.start,
              children: [
                Text(a.name, style: context.text.titleSmall),
                Text(isCard ? 'Outstanding' : 'Balance',
                    style: context.text.bodySmall?.copyWith(color: t.textMid)),
              ],
            ),
          ),
          Column(
            crossAxisAlignment: CrossAxisAlignment.end,
            children: [
              Text(Money.format(shown),
                  style: context.text.titleSmall?.copyWith(
                      color: isCard && balance.balance > 0 ? t.danger : t.textHigh)),
              if (isCard)
                Pressable(
                  onTap: () => Navigator.push(context,
                      MaterialPageRoute(builder: (_) => PayCreditCardScreen(cardId: a.id))),
                  child: Padding(
                    padding: const EdgeInsets.only(top: 2),
                    child: Text('Pay bill',
                        style: TextStyle(color: t.accentA, fontWeight: FontWeight.w600, fontSize: 13)),
                  ),
                ),
            ],
          ),
        ],
      ),
    );
  }
}

class _PendingCard extends StatelessWidget {
  final Txn txn;
  final VoidCallback onConfirm;
  final VoidCallback onDismiss;
  const _PendingCard({required this.txn, required this.onConfirm, required this.onDismiss});

  @override
  Widget build(BuildContext context) {
    final t = context.tokens;
    final isIncome = txn.type == TxnType.income;
    return GlassCard(
      onTap: onConfirm,
      strong: true,
      padding: const EdgeInsets.all(Insets.sm),
      radius: Corners.md,
      child: Row(
        children: [
          IconBadge(
              icon: Icons.sms_rounded, color: t.accentA, size: 42),
          const SizedBox(width: Insets.sm),
          Expanded(
            child: Column(
              crossAxisAlignment: CrossAxisAlignment.start,
              children: [
                Text(txn.merchant.isNotEmpty ? txn.merchant : 'Unknown',
                    style: context.text.titleSmall, maxLines: 1, overflow: TextOverflow.ellipsis),
                Text('${Dates.dateTime(txn.dateTime)} · tap to categorize',
                    style: context.text.bodySmall?.copyWith(color: t.textMid),
                    maxLines: 1, overflow: TextOverflow.ellipsis),
              ],
            ),
          ),
          Text('${isIncome ? '+' : '-'}${Money.short(txn.amount)}',
              style: context.text.titleSmall
                  ?.copyWith(color: isIncome ? t.success : t.danger)),
          IconButton(
            onPressed: onDismiss,
            icon: Icon(Icons.close_rounded, color: t.textMid, size: 20),
            visualDensity: VisualDensity.compact,
          ),
        ],
      ),
    );
  }
}
